### ISSUE TITLE
How to set variable with same name to different environments?

### ISSUE THREAD

**Issue by glensc** [gitlabform/gitlabform#411]
```yml
# Requires GITLAB_URL and GITLAB_TOKEN to be set as env variables
config_version: 3

gitlab:
  ssl_verify: true

projects_and_groups:
  "*":
    # https://gitlabform.github.io/gitlabform/reference/ci_cd_variables/
    # https://docs.gitlab.com/ee/api/project_level_variables.html#create-variable
    variables:
      S3_ACCESS_KEY-test/ee:
        key: S3_ACCESS_KEY
        value: "S3_ACCESS_KEY"
        masked: true
        environment_scope: test/ee
        filter[environment_scope]: test/ee

      S3_ACCESS_KEY-test/lv:
        key: S3_ACCESS_KEY
        value: "S3_ACCESS_KEY"
        masked: true
        environment_scope: test/lv
        filter[environment_scope]: test/lv

      S3_ACCESS_KEY-test/lt:
        key: S3_ACCESS_KEY
        value: "S3_ACCESS_KEY"
        masked: true
        environment_scope: test/lt
        filter[environment_scope]: test/lt

# vim:ts=2:sw=2:et
```

this results error:

```
Error: Entities S3_ACCESS_KEY-test/ee and S3_ACCESS_KEY-test/lv in variables for glen/playground are the same in terms of their defining keys: 'key'
```

**Comment by gdubicki** [comment]
Hi @glensc, thanks for reporting this! I think I managed to get it fixed in [3.1.2](https://github.com/gitlabform/gitlabform/releases/tag/v3.1.2), please check that version out and reopen if you still have any issues. :)

**Comment by glensc** [comment]
Confirming OK!
